### PR TITLE
mv erp test to compare two

### PR DESCRIPTION
**Pull request by jedwards4b**
Rewrite erp test using the compare two paradigm

Test suite: scripts_regression_tests.py, ERP_Ln9.f09_f09_mg17.F2000_DEV.cheyenne_intel.cam-outfrq9s
Test baseline: cesm2_0_alpha07d
Test namelist changes: 
Test status: bit for bit

Addresses #1647 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: config/cesm/config_archive.xml [hunk 1]
```diff
@@ -78,11 +78,11 @@
     <hist_file_extension>\.h.*.nc$|\.d[dovt]\.</hist_file_extension>
     <rest_history_varname>unset</rest_history_varname>
     <rpointer>
-      <rpointer_file>rpointer.ocn.restart$NINST_STRING</rpointer_file>
+      <rpointer_file>rpointer.ocn$NINST_STRING.restart</rpointer_file>
```
Comment (jedwards4b) [inline config/cesm/config_archive.xml]: This is unrelated to this PR, its a fix for test IRT.f09_g17.B1850

File: scripts/lib/CIME/SystemTests/erp.py [hunk 2]
```diff
@@ -8,132 +8,72 @@
...
+        # decomposition information in env_build.xml that
+        # needs to be regenerated for the above new tasks and thread counts
+        case_setup(self._case, test_mode=True, reset=True)
+
+    def _case_one_custom_postrun_action(self):
+        rundir1 = self._case1.get_value("RUNDIR")
+        rundir2 = self._case2.get_value("RUNDIR")
+        case = self._case1.get_value("CASE")
+        datenames = _get_datenames(self._case1)
+        for file_ in glob.iglob(os.path.join(rundir1,"*")):
```
Comment (jedwards4b) [inline scripts/lib/CIME/SystemTests/erp.py]: This copies the rpointer files and links the restart and hist restart files from the restart time to the case2 run directory.

Comment (billsacks) [reply scripts/lib/CIME/SystemTests/erp.py]: Thank you for coming up with this solution that avoids running the full short-term archiver.

Could the bulk of this function be put in some shared location so that it can be reused by other tests if needed? This could go in `scripts/lib/CIME/SystemTests/test_utils/`.

File: scripts/lib/CIME/SystemTests/erp.py [hunk 3]
```diff
@@ -8,132 +8,72 @@
...
+        # Note, some components, like CESM-CICE, have
+        # decomposition information in env_build.xml that
+        # needs to be regenerated for the above new tasks and thread counts
+        case_setup(self._case, test_mode=True, reset=True)
+
+    def _case_one_custom_postrun_action(self):
+        rundir1 = self._case1.get_value("RUNDIR")
+        rundir2 = self._case2.get_value("RUNDIR")
+        case = self._case1.get_value("CASE")
+        datenames = _get_datenames(self._case1)
```
Comment (billsacks) [inline scripts/lib/CIME/SystemTests/erp.py]: `_get_datenames` should have the leading underscore removed since it is no longer private.

File: scripts/lib/CIME/SystemTests/erp.py [hunk 4]
```diff
@@ -8,132 +8,72 @@
...
+        for file_ in glob.iglob(os.path.join(rundir1,"*")):
+            logger.info("File is {}".format(file_))
+            if os.path.basename(file_).startswith("rpointer"):
+                logger.info("Copy {} to {}".format(file_, rundir2))
+                shutil.copy(file_, rundir2)
+            elif os.path.basename(file_).startswith(case) and datenames[0] in file_:
```
Comment (billsacks) [inline scripts/lib/CIME/SystemTests/erp.py]: Is this guaranteed to pick up everything that's needed? There's a lot of complexity in `_archive_restarts` that isn't captured here. For the most part, I don't see what this might be missing, except that it seems not to capture any unfinished history files that the short-term archiver would capture with `get_histfiles_for_restarts`.

(This is why I was hoping we could still reuse pieces of the case_st_archive code....)

**Comment by jedwards4b** [comment]
@billsacks i've added your suggestions.

**Comment by billsacks** [comment]
@jedwards4b I'm playing with this a bit to see if I can accomplish the refactor I was envisioning of case_st_archive, in order to remove the test-specific logic you've added. I'll let you know if I have any luck.

**Comment by billsacks** [comment]
With the current code, plugged into

https://svn-ccsm-models.cgd.ucar.edu/clm2/trunk_tags/clm4_5_16_r253

I get a failure in this test: `ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default`:

```
2017-08-30 09:38:05: Exception during run:
`/glade/scratch/sacks/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp/run/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp.clm2.h0.0001-01-04-00000.nc` and `/glade/scratch/sacks/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp/case2/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp/run/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp.clm2.h0.0001-01-04-00000.nc` are the same file
Traceback (most recent call last):
  File "/glade2/scratch2/sacks/cesm_code/clm4_5_16_r253_newCime/cime/scripts/Tools/../../scripts/lib/CIME/SystemTests/system_tests_common.py", line 148, in run
    self.run_phase()
  File "/glade2/scratch2/sacks/cesm_code/clm4_5_16_r253_newCime/cime/scripts/Tools/../../scripts/lib/CIME/SystemTests/system_tests_compare_two.py", line 213, in run_phase
    self._case_one_custom_postrun_action()
  File "/glade2/scratch2/sacks/cesm_code/clm4_5_16_r253_newCime/cime/scripts/Tools/../../scripts/lib/CIME/SystemTests/erp.py", line 65, in _case_one_custom_postrun_action
    self.setup_restart()
  File "/glade2/scratch2/sacks/cesm_code/clm4_5_16_r253_newCime/cime/scripts/Tools/../../scripts/lib/CIME/SystemTests/system_tests_compare_two.py", line 264, in setup_restart
    shutil.copy(os.path.join(rundir1,histfile), rundir2)
  File "/usr/lib64/python2.7/shutil.py", line 119, in copy
    copyfile(src, dst)
  File "/usr/lib64/python2.7/shutil.py", line 69, in copyfile
    raise Error("`%s` and `%s` are the same file" % (src, dst))
Error: `/glade/scratch/sacks/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp/run/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp.clm2.h0.0001-01-04-00000.nc` and `/glade/scratch/sacks/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp/case2/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp/run/ERP_P24x2_Ld5.f10_f10.I1850Clm50Bgc.cheyenne_intel.clm-default.20170830_091616_gns0xp.clm2.h0.0001-01-04-00000.nc` are the same file
```

It's fine with me if you wait to look into this until I've gotten a chance to play around with a possible refactor that might make this issue irrelevant.

**Comment by jedwards4b** [comment]
This is a case where the history file has the restart date in it and so is linked to rundir2, we need to add code to specifically exclude history files from the link step and only copy them if needed in the next step.
